test/unit/theory/theory_bv_rewriter_white.h

```diff
@@ -100,6 +102,29 @@ class TheoryBvRewriterWhite : public CxxTest::TestSuite
     TS_ASSERT_EQUALS(nr, Rewriter::rewrite(nr));
   }
 
+  void testRewriteAshrEliminate()
+  {
+    TypeNode bvType = d_nm->mkBitVectorType(4);
+    Node zero = d_nm->mkConst(BitVector(4, 0u));
+    Node one = d_nm->mkConst(BitVector(4, 1u));
+    Node three = d_nm->mkConst(BitVector(4, 3u));
+    Node fifteen = d_nm->mkConst(BitVector(4, 15u));
+
+    Node fifteen_ashr_one = d_nm->mkNode(BITVECTOR_ASHR, fifteen, one);
+    Node three_ashr_one = d_nm->mkNode(BITVECTOR_ASHR, three, one);
+    Node three_lshr_one = d_nm->mkNode(BITVECTOR_LSHR, three, one);
+    TS_ASSERT(RewriteRule<AshrEliminate>::applies(fifteen_ashr_one));
+    TS_ASSERT(RewriteRule<AshrEliminate>::applies(three_ashr_one));
+    TS_ASSERT(!RewriteRule<AshrEliminate>::applies(three_lshr_one));
+    Node fifteen_ashr_one_rw =
+        RewriteRule<AshrEliminate>::apply(fifteen_ashr_one);
+    Node three_ashr_one_rw = RewriteRule<AshrEliminate>::apply(three_ashr_one);
+    TS_ASSERT_EQUALS(Rewriter::rewrite(fifteen_ashr_one),
+                     Rewriter::rewrite(fifteen_ashr_one_rw));
+    TS_ASSERT_EQUALS(Rewriter::rewrite(three_ashr_one),
+                     Rewriter::rewrite(three_ashr_one_rw));
```

Comment:
Not sure how meaningful this test is. Would it make more sense to do a case split on the MSB and check the following two cases:
```
(rewrite (bvashr (concat #b1 s) t)) == (bvnot (bvlshr (bvnot (concat #b1 s)) t))
(rewrite (bvashr (concat #b0 s) t)) == (bvlshr (concat #b0 s) t)
```

Reply:
Wouldn't the test then be too similar to the rewrite itself? What if we decide at some point to eliminate `bvashr` in some other manner? Then the test will break. Do you think it makes sense to do as you say, and then change the test if we ever change the elimination code?

Reply:
What would be actually better is to make a sat call and check the equivalence of the original and the rewritten node. But I'm not sure if this is overkill here. You can leave it for now.